evalml/pipelines/pipeline_base.py

```diff
@@ -205,7 +205,8 @@ def _compute_features_during_fit(self, X, y):
         return X_t
 
     def _fit(self, X, y):
-
+        X.reset_index(inplace=True, drop=True)
+        y.reset_index(inplace=True, drop=True)
```

Comment:
Include this code in order to avoid errors in `TargetEncoder` when used with the `Imputer`

Reply:
mmm I agree. Yeah, I can make that change, thanks for the catch!